tests/test_utils.py

```diff
@@ -303,126 +303,15 @@ def test_expand_three_nonconsecutive_nonascending_wires(self, tol):
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
 
-class TestRecorder:
```

Comment:
Minor suggestion here (not too crucial): small unit tests for the behaviour of the `QueuingContext` could be included. An example could be creating a test class, that inherits from `QueuingContext` and checking that each of the standard functionalities in there indeed works the way they should (e. g. `remove_operator` indeed throwing an error in certain cases).

Overall this would more so be a sanity check and an indicator that there is something going wrong there, should parts of the logic change.

Reply:
Yes, that is totally right. I'll add some tests!